data/components/frame-table.js

```diff
@@ -125,8 +125,9 @@ var FrameRow = React.createFactory(React.createClass({
    * This is an optimization that makes the list rendering a lot faster.
    */
   shouldComponentUpdate: function(nextProps, nextState) {
-    return this.props.frame == nextProps.selection ||
-      this.props.frame == this.props.selection;
+    return nextProps.selection !== this.props.selection && (
+      this.props.frame == nextProps.selection ||
+      this.props.frame == this.props.selection);
```

Comment:
I had to add `nextProps.selection !== this.props.selection` here because otherwise it would re-render the selected element on every new frame, and doing so closes the expanded tree